tools/automated-dns-for-instances/main.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud import dns
+import time
+import re
+
+PROJECT_ID=<PROJECT_ID>
+ZONE=<ZONE_ID>
+DOMAIN=<DOMAIN> # should have a trailing dot
```

Comment:
Can we make this a "template file" that asserts that domain ends with dot?
e.g. make this
```
PROJECT_ID=${PROJECT_ID}
ZONE=${ZONE_ID}
DOMAIN=${DOMAIN}  # should have a trailing dot
```
and use something like [`enbsubst`](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) in a small shell script. or wrap this in a small terraform module for deployment and use [`template_file`](https://www.terraform.io/docs/providers/template/d/file.html)